managed/resource_telemetry_provider.go

```diff
@@ -0,0 +1,547 @@
+/*
+ * Copyright © 2022-present Yugabyte, Inc. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package managed
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/schemavalidator"
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	openapiclient "github.com/yugabyte/yugabytedb-managed-go-client-internal"
+)
+
+type resourceTelemetryProviderType struct{}
+
+func (r resourceTelemetryProviderType) GetSchema(_ context.Context) (tfsdk.Schema, diag.Diagnostics) {
+	return tfsdk.Schema{
+		Description: `The resource to create a telemetry provider config in YugabyteDB Managed.`,
+		Attributes: map[string]tfsdk.Attribute{
+			"account_id": {
+				Description: "The ID of the account this telemetry provider config belongs to.",
+				Type:        types.StringType,
+				Computed:    true,
+			},
+			"project_id": {
+				Description: "The ID of the project this telemetry provider config belongs to.",
+				Type:        types.StringType,
+				Computed:    true,
+			},
+			"config_id": {
+				Description: "The ID of the telemetry provider config.",
+				Type:        types.StringType,
+				Computed:    true,
+			},
+			"config_name": {
+				Description: "The name of the telemetry provider configuration",
+				Type:        types.StringType,
+				Required:    true,
+			},
+			"type": {
+				Description: "Defines different exporter destination types. ",
+				Type:        types.StringType,
+				Required:    true,
+				Validators:  []tfsdk.AttributeValidator{stringvalidator.OneOf("DATADOG", "GRAFANA", "SUMOLOGIC")},
+			},
+			"is_valid": {
+				Description: "Signifies whether the configuration is valid or not ",
```

Comment:
To you expect the user to put if the configuration is valid or not ??? Sounds crazy

Reply:
No I've marked it as `Computed: true` that makes it clear that user should not fill this up but it will be computed by the servers. Is that not how `Computed: true` field works?

Reply:
But should always be valid,  not sure I understand the point. 

Non-valid settings should never be in the state anyway no ?

Reply:
They can get invalid when the api key expires. There is a scheduled process that runs on the apiserver that sets this field. TF when performing the read operation can update it's state

Reply:
Ok, but still I don't understand the value there,  this information has value only if the user will use it in the future (using output for example). Not a blocker just don't see the value of this field.

Reply:
Perhaps a user can `update` or `delete` the resource when `is_valid` field becomes invalid?